apps/explorer/src/pages/transaction-result/TransactionData.tsx

```diff
@@ -0,0 +1,279 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { CoinFormat, useFormatCoin } from '@mysten/core';
+import {
+    getGasData,
+    getTotalGasUsed,
+    getTransactionKind,
+    getTransactionKindName,
+    type ProgrammableTransaction,
+    SUI_TYPE_ARG,
+    type SuiTransactionBlockResponse,
+} from '@mysten/sui.js';
+
+import { InputsCard } from '~/pages/transaction-result/programmable-transaction-view/InputsCard';
+import { TransactionsCard } from '~/pages/transaction-result/programmable-transaction-view/TransactionsCard';
+import { CopyToClipboard } from '~/ui/CopyToClipboard';
+import { DescriptionItem } from '~/ui/DescriptionList';
+import { Divider } from '~/ui/Divider';
+import { Heading } from '~/ui/Heading';
+import { CheckpointSequenceLink, ObjectLink } from '~/ui/InternalLink';
+import { Text } from '~/ui/Text';
+import {
+    TransactionBlockCard,
+    TransactionBlockCardSection,
+} from '~/ui/TransactionBlockCard';
+
+function GasAmount({
+    amount,
+    isHighlighted,
+}: {
+    amount?: bigint | number;
+    isHighlighted?: boolean;
+}) {
+    const [formattedAmount, symbol] = useFormatCoin(
+        amount,
+        SUI_TYPE_ARG,
+        CoinFormat.FULL
+    );
+
+    return (
+        <div className="flex flex-wrap gap-1">
+            <div className="flex flex-wrap items-center gap-1">
+                <Text
+                    variant="pBody/medium"
+                    color={isHighlighted ? 'success-dark' : 'steel-darker'}
```

Comment:
can we pull the textColor into a variable since it's re-used for this group of related text?